ql/src/java/org/apache/hadoop/hive/ql/metadata/PartitionIterable.java

```diff
@@ -173,6 +183,42 @@ public PartitionIterable(Hive db, Table table, Map<String, String> partialPartit
     this(db, table, partialPartitionSpec, batchSize, getColStats, false, null, null);
   }
 
+  public PartitionIterable(Hive db, Table table, int batchSize, GetPartitionsRequest getPartitionsRequest)
```

Comment:
`batchSize` should be the last param

Reply:
could we pass `Hive db, Table table` through the GetPartitionsRequest
````
public PartitionIterable(GetPartitionsRequest getPartitionsRequest, int batchSize) {
...
}
````

Reply:
We would need a `Hive db` to be passed on. We can definitely do without a `Table table`. 
```
public PartitionIterable(Hive db, GetPartitionsRequest getPartitionsRequest, int batchSize) {
...
}
```